src/logic/FetchTopicRemotes.ts

```diff
@@ -0,0 +1,275 @@
+ * @returns Array of search results (filtered through the blacklist)
+ */
+export async function getTaggedRepos(tag: RepoTopic, page = 1, BLACKLIST:string[] = [], query?: string) {
+  // www is needed or it will block with "cross-origin" error.
+  let url = query
+    ? `https://api.github.com/search/repositories?q=${encodeURIComponent(`${query}+topic:${tag}`)}&per_page=${ITEMS_PER_REQUEST}`
+    : `https://api.github.com/search/repositories?q=${encodeURIComponent(`topic:${tag}`)}&per_page=${ITEMS_PER_REQUEST}`;
+
+  // We can test multiple pages with this URL (58 results), as well as broken iamges etc.
+  // let url = `https://api.github.com/search/repositories?q=${encodeURIComponent("topic:spicetify")}`;
+  if (page) url += `&page=${page}`;
+  // Sorting params (not implemented for Marketplace yet)
+  // if (sortConfig.by.match(/top|controversial/) && sortConfig.time) {
+  //     url += `&t=${sortConfig.time}`
+  const allRepos = await fetch(url).then(res => res.json()).catch(() => []);
+  if (!allRepos.items) {
+    Spicetify.showNotification("Too Many Requests, Cool Down.");
+    return;
+  }
+  const filteredResults = {
+    ...allRepos,
+    // Include count of all items on the page, since we're filtering the blacklist below,
+    // which can mess up the paging logic
+    page_count: allRepos.items.length,
+    items: allRepos.items.filter(item => !BLACKLIST.includes(item.html_url)),
+  };
+
+  return filteredResults;
+}
+
+// TODO: add try/catch here?
+// TODO: can we add a return type here?
+/**
+* Get the manifest object for a repo
+* @param user Owner username
+* @param repo Repo name
+* @param branch Default branch name (e.g. main or master)
+* @returns The manifest object
+*/
+async function getRepoManifest(user: string, repo: string, branch: string) {
+  const sessionStorageItem = window.sessionStorage.getItem(`${user}-${repo}`);
+  const failedSessionStorageItems = window.sessionStorage.getItem("noManifests");
+  if (sessionStorageItem) return JSON.parse(sessionStorageItem);
+
+  const url = `https://raw.githubusercontent.com/${user}/${repo}/${branch}/manifest.json`;
+  if (failedSessionStorageItems?.includes(url)) return null;
+
+  const manifest = await fetch(url).then(res => res.json()).catch(
+    () => addToSessionStorage([url], "noManifests"),
+  );
+  if (manifest) window.sessionStorage.setItem(`${user}-${repo}`, JSON.stringify(manifest));
+
+  return manifest;
+}
+
+// TODO: can we add a return type here?
+/**
+* Fetch extensions from a repo and format data for generating cards
+* @param contents_url The repo's GitHub API contents_url (e.g. "https://api.github.com/repos/theRealPadster/spicetify-hide-podcasts/contents/{+path}")
+* @param branch The repo's default branch (e.g. main or master)
+* @param stars The number of stars the repo has
+* @returns Extension info for card (or null)
+*/
+export async function fetchExtensionManifestFromTopic(contents_url: string, branch: string, stars: number, hideInstalled = false) {
```

Comment:
I'd rather not have so much of this code duplicated with the mono-manifest stuff. That's part of the reason why I didn't really want to support both in the first place.

Reply:
As far as this goes, it more or less just comes down to having two files(which we have) and adding a few if statements in select locations.  I've already worked out the system so there's not much else that needs to be done other than adding extension support to the mono manifest system.